drivers/gpio/gpio_rcar.c

```diff
@@ -48,51 +46,50 @@ struct gpio_rcar_data {
 #define OUTDTSEL 0x40   /* Output Data Select Register */
 #define BOTHEDGE 0x4c   /* One Edge/Both Edge Select Register */
 
-static inline uint32_t gpio_rcar_read(const struct gpio_rcar_cfg *config,
+static inline uint32_t gpio_rcar_read(const struct device *dev,
 				      uint32_t offs)
 {
-	return sys_read32(config->reg_addr + offs);
+	return sys_read32(DEVICE_MMIO_GET(dev) + offs);
```

Comment:
Please use `DEVICE_MMIO_NAMED_GET()` instead. You might find some inspiration in `drivers/gpio/gpio_intel.c`.